cnf-tests/testsuites/e2esuite/dpdk/dpdk.go

```diff
@@ -173,6 +178,107 @@ var _ = Describe("dpdk", func() {
 		}
 	})
 
+	Context("vhostnet", func() {
+		var policyHasVhostnet bool
+		execute.BeforeAll(func() {
+			if !discovery.Enabled() {
+				CleanSriov()
+				createSriovPolicyAndNetworkDPDKOnlyWithVhost()
+			} else {
+				sriovNetworkNodePolicyList := &sriovv1.SriovNetworkNodePolicyList{}
+				err := client.Client.List(context.TODO(), sriovNetworkNodePolicyList)
+				Expect(err).ToNot(HaveOccurred())
+				for _, policy := range sriovNetworkNodePolicyList.Items {
+					if policy.Spec.ResourceName == dpdkResourceName && policy.Spec.NeedVhostNet {
+						policyHasVhostnet = true
+					}
+				}
+			}
+		})
+
+		BeforeEach(func() {
+			if discovery.Enabled() && !policyHasVhostnet {
+				Skip("Missing SriovNetworkNodePolicy with NeedVhostNet enabled")
+			}
+		})
+		Context("Client should be able to forward packets", func() {
+			It("Should be able to transmit packets", func() {
+				var out string
+				var err error
+
+				command := CREATE_TAP_DEVICE_COMMAND + `
+				dpdk-testpmd --vdev net_tap0,iface=tap23 --no-pci -- -ia --forward-mode txonly
+				sleep INF
+				`
+				txDpdkWorkloadPod, err := createDPDKWorkload(nodeSelector,
+					command,
+					false,
+					[]corev1.Capability{"NET_ADMIN"},
+					dpdkWorkloadMac,
+				)
+				Expect(err).ToNot(HaveOccurred())
+
+				By("Parsing output from the client DPDK application")
+				Eventually(func() string {
+					out, err = pods.GetLog(txDpdkWorkloadPod)
+					Expect(err).ToNot(HaveOccurred())
+					return out
+				}, 2*time.Minute, 1*time.Second).Should(ContainSubstring(LOG_ENTRY),
+					"Cannot find accumulated statistics")
+				By("Checking the tx output from the client DPDK application")
+				checkTxOnly(out)
```

Comment:
can you please also check the tap side here

Reply:
done